frontend/src/entities/jobs/algorithms.tsx

```diff
@@ -71,83 +73,114 @@ export type Hyperparameter = {
     converter?: (input: any) => any;
 };
 
-const floatValidator = (required = false, positive = false) => {
-    let number_context = z.number({
-        invalid_type_error: `Must be a float${required ? ' and is required.' : ''}`,
+const floatValidator = (fieldName: string, required = false, positive = false) => {
+    return numberValidator(fieldName, {
+        isFloat: true,
+        required: required,
+        min: positive ? 0 : Number.NEGATIVE_INFINITY,
+        includeMin: false // If positive, can't be zero, will never reach Number.NEGATIVE_INFINITY regardless
     });
+};
 
-    if (positive) {
-        number_context = number_context.min(0, { message: 'Must be at least 0' });
-    }
-
-    let context: any = z.preprocess(Number, number_context);
-    if (!required) {
-        context = context.optional().or(z.literal(''));
-    }
-    return context;
+const positiveIntValidator = (fieldName: string, required = true, minimum = 1, inputProps : rangeValidatorProps = {}) => {
```

Comment:
We may need to update the lint/formatter rule if this wasn't picked up.
```suggestion
const positiveIntValidator = (fieldName: string, required = true, minimum = 1, inputProps: rangeValidatorProps = {}) => {
```

Reply:
Yeah, something may have changed? I noticed that it doesn't automatically run during the local hosting like it used to or it isn't working as expected. I will try running a build to see what the difference is.

Reply:
Linter seems to not be working properly